data/campaigns/Descent_Into_Darkness/scenarios/03_A_Haunting_in_Winter.cfg

```diff
@@ -132,222 +111,986 @@
+        [/scroll_to]
+
+        {MODIFY_UNIT (id=S3_Rebel_Soul) side 1}
+        [delay]
+            time=400
+        [/delay]
+        {MODIFY_UNIT (id=S3_Rebel_Soul) side 2}
+        [delay]
+            time=400
+        [/delay]
+        {MODIFY_UNIT (id=S3_Rebel_Soul) side 1}
+        [delay]
+            time=400
+        [/delay]
+        {MODIFY_UNIT (id=S3_Rebel_Soul) side 2}
+        [delay]
+            time=400
+        [/delay]
+        {MODIFY_UNIT (id=S3_Rebel_Soul) side 1}
+        [delay]
+            time=400
+        [/delay]
+
+        [message]
+            speaker=Malin Keshar
+            message= _ "I think I’ve got it under control again. That was a close call. Let me try the other one now."
+        [/message]
+
+        [animate_unit]
+            [filter]
+                id=Malin Keshar
+            [/filter]
+            [primary_attack]
+                range=ranged
+            [/primary_attack]
+            flag=attack
+        [/animate_unit]
+
+        [delay]
+            time=400
+        [/delay]
+
+        [unit]
+            type=Ghost
+            x,y=$loc2.x,$loc2.y
+            side=1
+            animate=yes
+        [/unit]
+
+        [delay]
+            time=500
+        [/delay]
+
+        [message]
+            speaker=Darken Volk
+            message= _ "No mishaps this time, but you will need to... practice more."
+        [/message]
+
+        {CLEAR_VARIABLE loc1}
+        {CLEAR_VARIABLE loc2}
+
+        [terrain]
+            x=11,12
+            y=3,3
+            terrain=Re
+        [/terrain]
+        [terrain]
+            x=11,12
+            y=4,4
+            terrain=Uu
+        [/terrain]
+        [redraw][/redraw]
+
+        {FULL_HEAL (side=1)}
+
+        {OBJECTIVE_EXPLORE}
     [/event]
 
+    # Fifth area (scorpions/trolls)
     [event]
-        name=die
+        name=enter hex
         [filter]
-            side=4
-            canrecruit=yes
+            side=1
+            x,y=5-9,6-7
         [/filter]
 
-        # Regain control of "rebel" ghosts
-        {MODIFY_UNIT (side,canrecruit=4,no) side 1}
-        [redraw]
-            clear_shroud=yes
-        [/redraw]
+        [message]
+            speaker=Malin Keshar
+            message= _ "Rail tracks and mines... dwarves used to live here, but there do not seem to be any guards. I wonder where they went?"
+        [/message]
+    [/event]
+
+    [event]
+        name=sighted
+        [filter]
+            race=troll
+        [/filter]
+        [filter_second]
+            side=1
+        [/filter_second]
 
         [message]
             speaker=Malin Keshar
-            message= _ "I think I’ve got the rest of them under control now. That was a close call."
+            message= _ "Trolls! I suppose I should have expected this."
         [/message]
+
+        [message]
+            speaker=Darken Volk
+            message= _ "Threats to your skeletons, but no more than fodder for your ghosts. Eliminate them all."
```

Comment:
It's easy to trigger this before you get ghosts.

Reply:
I'll double check, but this should be impossible since you would have to travel through cave walls to get to the trolls (before you kill the nagas and get ghost). Part of the problem right now is that I forgot to push the new map...

Reply:
With the new map, all looks good (or evil, "good" isn't a fitting word for this storyline).